src/Core/ServiceWrapper/Main.cs

```diff
@@ -815,6 +819,25 @@ void Stop()
                 }
             }
 
+            void StopWait()
+            {
+                Log.Info("Stopping the service with id '" + descriptor.Id + "'");
+                if (s is null)
+                    ThrowNoSuchService();
+
+                if (s.Started)
+                    s.StopService();
+
+                while (s != null && s.Started)
+                {
+                    Log.Info("Waiting the service to stop...");

```

Comment:
Should we really print it on every cycle in the loop?

Reply:
@oleg-nenashev Well, I think it's useful that the wrapper shows that it's not completely stuck, but it's up to you

Reply:
I agree that we print something to indicate that the program isn't stuck. Later we can do something interesting such as drawing a progress bar in the console.